src/components/Admin/Admin.tsx

```diff
@@ -29,37 +29,37 @@ const Admin: React.FC = () => {
     {
       route: 'templates',
       header: strings.MENU_ITEM_ADMIN_TEMPLATES,
-      Element: () => <Templates />,
+      Element: <Templates />,
```

Comment:
This was what was causing the re-mounting-on-query-update problem. Very annoying, but glad to have fixed, as the same problem would be affecting everything under the "Admin" route, including the template builder.